core/src/main/scala/filodb.core/memstore/TimeSeriesShard.scala

```diff
@@ -429,6 +440,19 @@ class TimeSeriesShard(val ref: DatasetRef,
             case e: Exception                   => logger.error(s"Unexpected ingestion err", e); disableAddPartitions()
           }
         } else {
+          val oldTimestamp = groupTimestamps(group)
+          val newTimestamp = Math.max(oldTimestamp, ingestionTime) // monotonic clock
```

Comment:
This check should not happen within `onNext`, as the ingestion time does not change for each sample.  This makes the check much more expensive than it should be ... just check it after each call to ingest().

Reply:
Ingestion time can vary by group, and the monotonic adjustment shouldn't span groups.